ibis/common/geospatial.py

```diff
@@ -19,22 +19,38 @@ def _format_point_value(value: Iterable) -> str:
     return ' '.join(str(v) for v in value)
 
 
-def _format_linestring_value(value: Iterable) -> str:
+def _format_linestring_value(value: Iterable, nested=False) -> str:
     """Convert a iterable with a linestring to text."""
-    return ', '.join(_format_point_value(point) for point in value)
+    template = '({})' if nested else '{}'
+    # fix wrong structure
```

Comment:
Can you elaborate here? What exactly is wrong is about the structure if the first element is a `tuple` or `list`?

Reply:
for example this WKT `POLYGON ((30 10, 40 40, 20 40, 10 20, 30 10))`

if anyone tries that in python `(((30, 10), (40, 40), (20, 40), (10, 20), (30, 10)))` the outer parenthesis doesn't have any effect. the "correct" way to write that would be `(((30, 10), (40, 40), (20, 40), (10, 20), (30, 10)),)` (ensuring the tuple format on the outer parenthesis)

so this code is used to ensure the "correct" format of this data.

Reply:
I don't think we should try to mirror the syntax of postgres/postgis types here with a Python equivalent. A sequence of pairs of numbers is the contract.

Reply:
but it should be a tuple, another example shows that:

`POLYGON ((35 10, 45 45, 15 40, 10 20, 35 10), (20 30, 35 35, 30 20, 20 30))`

Reply:
I don't think we should be second guessing user input types. Can you write down the type of the input expected?

Reply:
`tuple(tuple(tuple()))` where the first is the structure of the polygon the second is the structure of the line and the third is each point(x, y)

so you suggestion is NOT change the structure (`value = (value,)`) and just check and raise an error?

Reply:
anymore thoughts about that? should I change anything? or is it already OK?

Reply:
ah and if I should raise an error .. what would be the best `Exception` to be used? `IbisInputError`?

Reply:
Yeah, I think that makes sense.

Reply:
sounds good. I will change that! thanks!

Reply:
done! let me know your thoughts about the message and the type hint.